js/objects/utils/serialization.js

```diff
@@ -85,6 +85,13 @@ class STDeserializer {
                     let targetView = document.querySelector(`[part-id="${this.targetId}"]`);
                     targetView.appendChild(rootView);
                     this.dispatchViewAdded(rootView);
+                    rootPart.sendMessage({
```

Comment:
Yeah this is where things get a little confusing for me. Wouldn't the sending of `newModel`, if not handled by the given part, delegate up to System and then result in the creation of an (extra) new model? It seems like this does need to be a separate message name for use at the Simpletalk level, like `newPart` or something

Reply:
i put the `shoultNotDelegate` flag there, so it won't go up the Stack 

The main reason I didn't add new message names is because of `deleteModel`. If you add some new name there like `deletePart` then when is this message sent? it has to go before any `deleteModel` is handled but then there is no guarantee it the `deletePart` handler will be interpreted and run before the model is deleted and view removed. So you need something smarter there that allows you to hold off deleting before the `deletePart` is done, which is possible but is not trivial. 

Then having something like `addPart` but `deleteModel` felt wrong and seemed impossible to remember. So I decided to let it be and stick with `addModel` but not have it go up the chain.

Reply:
Ok makes sense